src/endpointsApi.js

```diff
@@ -0,0 +1,149 @@
+'use strict'
+
+const logger = require('./logger')
+const EndpointModel = require('./models/endpoints')
+const {handleServerError, validateEndpoint} = require('./util')
+const KoaBodyParser = require('@viweei/koa-body-parser')
+
+const saveEndPoint = body => {
+  return new Promise((resolve, reject) => {
+    const endpoint = new EndpointModel(body)
+    endpoint.save(err => {
+      if (err) reject(err)
+      resolve(endpoint)
+    })
+  })
+}
+
+const updateEndPoint = (endpointId, body) => {
+  return new Promise((resolve, reject) => {
+    EndpointModel.findOneAndUpdate(
+      {_id: endpointId},
+      body,
+      {new: true, runValidators: true},
+      (err, result) => {
+        if (err) reject(err)
+        resolve(result)
+      }
+    )
+  })
+}
+
+const createCreateEndpoint = router => {
+  router.post('/endpoints', async (ctx, next) => {
+    const failureMsg = 'Endpoint creation/update failed: '
+
+    try {
+      await KoaBodyParser()(ctx, () => {})
+
+      if (
+        !ctx.request ||
+        !ctx.request.body ||
+        !Object.keys(ctx.request.body).length
+      ) {
+        ctx.status = 400
+        const err = `${failureMsg}Invalid endpoint object`
+        ctx.body = {error: err}
+        logger.error(err)
+        return next()
+      }
+      const body = Object.assign({}, ctx.request.body, {
+        lastUpdated: Date.now()
+      })
+      const inValid = validateEndpoint(body)
+
+      if (inValid) {
+        ctx.status = 400
+        ctx.error = {error: inValid}
+        logger.error(inValid)
+        return next()
+      }
+
+      await saveEndPoint(body)
+        .then(result => {
+          ctx.status = 201
+          ctx.body = result
+          logger.info(`Endpoint with id ${result.id} created`)
```

Comment:
@bradsawadye It might be useful to include the name of the endpoint in these logger messages. The IDs don't mean too much at first glance.

Reply:
done in https://github.com/jembi/openhim-mediator-mapping/pull/48/commits/04ed2093b4403bc8519d763fa232a2a383b8a870